tools/llhd-sim/llhd-sim.cpp

```diff
@@ -156,6 +157,7 @@ static LogicalResult applyMLIRPasses(ModuleOp module) {
   PassManager pm(module.getContext());
 
   pm.addPass(createConvertLLHDToLLVMPass());
+  pm.addPass(::mlir::createReconcileUnrealizedCastsPass());
 
```

Comment:
cc @maerhart

Reply:
I did not add the `reconcile-unrealized-casts` pass in to the `circt-lec` pipeline after `convert-verif-to-smt`.  I am guessing that eitherit  doesn't need the pass because the canonicalizer cleans these up, or we just don't have a test that is catching the issue.
```
  pm.addPass(createConvertVerifToSMT());
  pm.addPass(createSimpleCanonicalizerPass());
```

Reply:
Right now, I cannot think of a case where the canonicalizer is not enough. The cases I'm aware of in the LEC lowering are already dealt with in the pass itself.